internal/setup/setup.go

```diff
@@ -176,7 +200,13 @@ func (r *Release) validate() error {
 	// Check for glob conflicts.
 	for newPath, new := range globs {
 		for oldPath, old := range paths {
-			if new.Package == old.Package {
+			// Same entry, no conflict.
+			if new == old && newPath == oldPath {
+				continue
+			}
+			// Content extracted (not generated) from the same package can never
+			// be in conflict.
+			if new.Package == old.Package && new.Contents[newPath].Kind == GlobPath && old.Contents[oldPath].Kind != GeneratePath {
```

Comment:
This check doesn't seem sound. Why would we allow different slices to produce or refer to conflicting content inside the same directory?  The rule should be much more strict: if it's an overlaping generate path, either it's exactly the same or it's a conflict, because the actual generated content is undefined at this stage. This is unlike the normal logic for globs which are picking the same content from the same package.

Reply:
Let's discuss it in https://github.com/canonical/chisel/pull/143#discussion_r1671911925.